gems/railties/6.0/railties.rbs

```diff
@@ -1,3 +1,25 @@
+module Rails
+  module Configuration
+    class MiddlewareStackProxy
+      def insert_before: (*untyped args) ?{ () -> untyped } -> untyped
+
+      alias insert insert_before
+
+      def insert_after: (*untyped args) ?{ () -> untyped } -> untyped
+
+      def use: (*untyped args) ?{ () -> untyped } -> untyped
+    end
+  end
+end
+
+module Rails
+  class Engine
+    # Defines the routes for this engine. If a block is given to
+    # routes, it is appended to the engine.
+    def routes: () ?{ () -> untyped } -> untyped
```

Comment:
LGTM.

As mentioned in the above comment, the block is optional.
https://github.com/rails/rails/blob/v6.0.6.1/railties/lib/rails/engine.rb#L539

Reply:
Thanks! I believe the syntax I have used is marking the block as optional, unless you had a different syntax in mind or I've mistaken something?

Reply:
Sorry for the confusion. The second paragraph is a note for me.